### DISCUSSION TITLE
how to add smart thermostat as device

### DISCUSSION THREAD

**Discussion by romanbaranovic** [Koenkk/zigbee2mqtt#9568] [Devices]
i have this thermostat https://www.zigbee2mqtt.io/devices/TV01-ZB.html. i can add it as a device to my zigbee2mqtt configuration, but it is seen as unsuported device model TS0601. is there a way to force my zigbee2mqtt to recognize the device as TV01-ZB or i just have to setup external converter as it is described here https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#instructions?

**Comment by dvovla** [comment]
I'm not an expert, but I think this TRV should be supported in the current version of Z2M.  I bought the same TRV sold under the TESLA brand (it has a different Zigbee Manufacturer code) and I had the same problem.  It occurred to me to take an external converter and change the "Zigbee Manufacturer" parameter in it.  And it works except for updating the current temperature and maybe the battery status, I don't know yet.  These shortcomings may have already been resolved, but I have no idea how to fix them yet.

I used this convertor: https://gist.github.com/geoffreylagaisse/bb90214dbaf687465e5841f49685cf66
